### PR TITLE
Update indexing for dplyr_cummean (#5287)

### PR DESCRIPTION
**Pull request by brianrice2**
Indexing off by one, causing first index to be repeated twice (and last index to be dropped). My intuition is we must keep *p_out and *p_x synced, so need to increment both on line 108. 

Example given:
x <- 1:5
Expected: 1.0 1.5 2.0 2.5 3.0
Actual:  1.000000 1.000000 1.333333 1.750000 2.200000
PR output: 1.0 1.5 2.0 2.5 3.0

Tested locally using Rcpp. Feedback welcome.

### CODE REVIEW

**Comment by hadley** [comment]
Fixes #5287

**Comment by hadley** [comment]
Could you please add a unit test so we don't accidentally break this again?

**Comment by brianrice2** [comment]
Yes, good suggestion. I added an additional commit to check that `cummean()` is consistent with `cumsum() / seq_along()`, so either method produces the same result.

File: tests/testthat/test-window.R [hunk 1]
```diff
@@ -87,6 +87,11 @@ test_that("cummean is not confused by FP error (#1387)", {
   expect_true(all(cummean(a) == a))
 })
 
+test_that("cummean is consistent with cumsum() and seq_along() (#5287)", {
+  x <- 1:5
+  expect_equal(cummean(x), cumsum(x) / seq_along(x))
```
Comment (hadley) [inline tests/testthat/test-window.R:92]: Would you mind inlining the exact values here? That makes it a bit easier to understand in the future.

Comment (brianrice2) [reply tests/testthat/test-window.R:94]: Yep, added new changes. Do they line up with what you mean, or still something else?

Comment (hadley) [reply tests/testthat/test-window.R:94]: Sorry, I meant putting them in the test — `expect_equal(cummean(x), c(1, 1.5, 2, 2.5, 3))`. That ensures that the test is still useful even if `cumsum()` or `seq_along()` had bugs.

Comment (brianrice2) [reply tests/testthat/test-window.R:94]: Ahhh yes, that makes a lot of sense. Thanks for explaining. Agree it's better that way!

**Comment by romainfrancois** [comment]
I've simplified the code so that the loop starts at 0, not 1. There probably was undefined behavior for length 0 inputs.

**Comment by romainfrancois** [comment]
Thanks.

**Pull request merged**